### PR TITLE
Wrong use of ull for develop

### PR DESCRIPTION
**Pull request by nijtmans**
As promised in #449 : This is the same PR as already merged to support/1.x, but now cherry-picked to "develop"

### CODE REVIEW

File: demo/test.c [hunk 1]
```diff
@@ -223,9 +223,9 @@ static int test_mp_get_set_i64(void)
 
    DOR(mp_init(&a));
 
-   check_get_set_i64(&a, 0LL);
```
Comment (minad) [inline demo/test.c]: Here the cast is unnecessary. I would prefer if we only add casts where needed, e.g., in code like `(uint64_t)1 << ...`

Comment (nijtmans) [reply demo/test.c]: Agreed. Just as the "LL" postfix shouldn't have been here originally in the first place ;-)

File: demo/test.c [hunk 2]
```diff
@@ -637,8 +637,8 @@ static int test_mp_get_u32(void)
       printf("\nmp_get_u32() bad result!");
       goto LBL_ERR;
    }
-   mp_set_ul(&a, 0xFFFFFFFFuL);
-   if (mp_get_u32(&a) != 0xFFFFFFFFuL) {
+   mp_set_ul(&a, (unsigned long)UINT32_MAX);
```
Comment (minad) [inline demo/test.c]: there are more unnecessary casts, for example this one.

File: demo/test.c [hunk 3]
```diff
@@ -625,7 +625,7 @@ static int test_mp_get_u32(void)
    DOR(mp_init_multi(&a, &b, NULL));
 
    for (i = 0; i < 1000; ++i) {
-      t = (unsigned long)rand_long() & 0xFFFFFFFFuL;
+      t = (unsigned long)rand_long() & (unsigned long)UINT32_MAX;
```
Comment (minad) [inline demo/test.c]: x

File: demo/test.c [hunk 4]
```diff
@@ -2177,7 +2177,7 @@ static int test_mp_read_write_ubin(void)
 
    size = mp_ubin_size(&a);
    printf("mp_to_ubin_size  %zu - ", size);
-   buf = malloc(sizeof(*buf) * size);
+   buf = (unsigned char *)malloc(sizeof(*buf) * size);
```
Comment (minad) [inline demo/test.c]: I think we use uint8_t* here now

File: demo/timing.c [hunk 5]
```diff
@@ -47,7 +47,7 @@ static unsigned long lfsr = 0xAAAAAAAAuL;
 static unsigned int lbit(void)
 {
    if ((lfsr & 0x80000000uL) != 0uL) {
-      lfsr = ((lfsr << 1) ^ 0x8000001BuL) & 0xFFFFFFFFuL;
+      lfsr = ((lfsr << 1) ^ 0x8000001BuL) & (unsigned long)UINT32_MAX;
```
Comment (minad) [inline demo/timing.c]: x

File: s_mp_rand_jenkins.c [hunk 6]
```diff
@@ -28,7 +28,7 @@ static uint64_t s_rand_jenkins_val(void)
 void s_mp_rand_jenkins_init(uint64_t seed)
 {
    int i;
-   jenkins_x.a = 0xF1EA5EEDuLL;
+   jenkins_x.a = (uint64_t)0xF1EA5EEDuL;
```
Comment (minad) [inline s_mp_rand_jenkins.c]: x

File: s_mp_rand_jenkins.c [hunk 7]
```diff
@@ -42,7 +42,7 @@ mp_err s_mp_rand_jenkins(void *p, size_t n)
       int i;
       uint64_t x = s_rand_jenkins_val();
       for (i = 0; (i < 8) && (n > 0u); ++i, --n) {
-         *q++ = (char)(x & 0xFFuLL);
+         *q++ = (char)(x & (uint64_t)0xFFu);
```
Comment (minad) [inline s_mp_rand_jenkins.c]: x

File: demo/test.c (hunk 1 above)
Comment (minad) [reply demo/test.c]: Yes, unfortunately fperrad is overdoing it. You've seen me probably complaining about the linting before ;)

**Comment by nijtmans** [comment]
I think all those unnecessary type-cast are gone now.

**Comment by minad** [comment]
@sjaeckel This is ready to merge

**Comment by sjaeckel** [comment]
can we somehow make sure that this doesn't happen again in the future?

**Comment by minad** [comment]
Actually through the c89-c99 roundtrip some notations are fixed. Besides we might want to investigate if there is a better possibility for the linting in the CI. Furthermore we could add rules to this helper.pl checker. But then I don't think too much harm is done with such things, these are mostly stylistic things and only rarely lead to bugs.

**Pull request merged**